src/bp/core/services/telemetry/sdk-methods.ts

```diff
@@ -0,0 +1,194 @@
+    this.lock = 'botpress:telemetry-SDK'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'custom_roles',
+      event_data: { schema: '1.0.0', SDKMethods: await this.getSDKUsage() }
+    }
+  }
+
+  private async getSDKUsage(): Promise<SDKUsageEvent> {
+    const bots = await this.botService.getBotsIds()
+    return { actions: await this.getActionUsages(bots), hooks: await this.getHooksUsages() }
+  }
+
+  private async getActionUsages(bots: string[]): Promise<SDKUsage> {
+    const rootFolder = 'actions'
+    const globalActionsNames = await this.ghostService
+      .global()
+      .directoryListing('/', `${rootFolder}/*.js`)
+      .map(path => path.split('/').pop() || '')
+
+    const reducer = async (parsedFilesAcc, botId) => {
+      try {
+        const botActionsNames = await this.ghostService.forBot(botId).directoryListing(`/${rootFolder}`, '*.js')
+        const parsedFiles = await Promise.map(botActionsNames, name =>
+          this.parseFile(name, `/${rootFolder}`, { botId })
+        )
+        return [...parsedFilesAcc, ...parsedFiles]
+      } catch (error) {
+        return []
+      }
+    }
+
+    const global = await Promise.map(globalActionsNames, name => this.parseFile(name, `/${rootFolder}`))
+    const perBots = await Promise.reduce(bots, reducer, [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private async getHooksUsages(): Promise<SDKUsage> {
+    const reducer = async (parsedFilesAcc, botHooks: BotHooks) => {
+      const { botId, hooks } = botHooks
+
+      try {
+        const parsedFiles = await this.parseHooks(
+          hooks.filter(hook => hook.type === 'custom'),
+          botId
+        )
+        return [...parsedFilesAcc, ...parsedFiles]
+      } catch (err) {
+        return []
+      }
+    }
+
+    const hooksPayload = await getHooksLifecycle(this.botService, this.ghostService, false)
+    const global = await this.parseHooks(hooksPayload.global.filter(hook => hook.type === 'custom'))
+    const perBots = await Promise.reduce(hooksPayload.perBots, reducer, [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private async parseHooks(hooks: Hook[], botId?: string): Promise<ParsedFile[]> {
+    return Promise.map(hooks, async hook => {
+      const { lifecycle, enabled, path } = hook
+      const usageParams: UsageParams = { lifecycle, enabled }
+
+      if (botId) {
+        usageParams.botId = botId
+      }
+
+      return this.parseFile(path!, '/hooks', usageParams)
+    })
+  }
+
+  private async parseFile(name: string, rootFolder: string, usageParams?: UsageParams): Promise<ParsedFile> {
+    const file = await this.readFileAsString(rootFolder, name, usageParams?.botId)
+    const functions = this.extractFunctions(parse(file, PARSE_CONFIG))
+    const usage: ParsedFile = { fileName: name.split('/').pop() || '', usages: this.parseMethods(functions) }
```

Comment:
Filenames are not obfuscated

Reply:
fixed: 37ab2dab6c341c1635105210ef047558706364bd